### PR TITLE
Remove dependency on Quick/Nimble

### PR DESCRIPTION
**Pull request by yoichitgy**
This PR addresses the same issue of [Swinject Issue #472](https://github.com/Swinject/Swinject/issues/472), and performs the migration from Quick/Nimble to XCTest in the same way as [Swinject PR #473](https://github.com/Swinject/Swinject/pull/473).

- Replaced `XxxSpec` using Quick/Nimble with `XxxTests` using XCTest.
- Removed dependency on Quick/Nimble. (We still need Carthage because of the dependency on Swinject.)

**Note**
Unit tests corresponding to those failed with Quick/Nimble still fail for macOS target. We need investigation later.
Screen Shot

### CODE REVIEW

File: Tests/iOS-tvOS/SwinjectStoryboardSpec.swift [hunk 1]
```diff
@@ -1,200 +0,0 @@
...
-            }
-            
-            afterEach {
-                SwinjectStoryboard.defaultContainer.removeAll()
-            }
-        }
-        // We need to have test bundle deployment target on iOS 9.0 in order to compile storyboards with references.
-        // However, we need to disable these tests when running on iOS <9.0
-        // Using #available(iOS 9.0, *) produces compiler warning for the reasons above
-        if ProcessInfo().isOperatingSystemAtLeast(OperatingSystemVersion(majorVersion: 9, minorVersion: 0, patchVersion: 0)) {
```
Comment (yoichitgy) [inline Tests/iOS-tvOS/SwinjectStoryboardSpec.swift:144]: We can remove the iOS version check because we no longer support iOS 8.

Comment (yoichitgy) [reply Tests/iOS-tvOS/SwinjectStoryboardSpec.swift:144]: Although the current iOS deployment target is set to 12.0, it looks the target was increased by accident in the following PR. It looks good to revert the target to 9.0 later.
https://github.com/Swinject/SwinjectStoryboard/pull/156#pullrequestreview-698432984

**Comment by mpdifran** [comment]
Do the failing tests also fail for `macOS` on the current master (using Quick and Nimble)? If so, that likely indicates that our library is broken on `macOS` for some use cases.

**Comment by yoichitgy** [comment]
The failing tests also fail for `macOS` using Quick and Nimble for the same conditions.

The tests fail with `nil` where a value is expected. I'm assuming Storyboard setting or some settings are broken in the `macOS` configuration of SwinijectStoryboard.

**Comment by mpdifran** [comment]
Gotcha. Are you cool with merging this PR after I do the release? Or do you think it's safe enough to go into the release?

**Review by mpdifran — APPROVED** [review]

**Comment by yoichitgy** [comment]
I don't mind merging this PR after you make the release👍 Meanwhile, I'll check the cause of the test failure more😉

**Comment by yoichitgy** [comment]
I noticed unit tests failed if a vew/window controller was instantiated with `instantiateInitialController`, but still don't know why the test fail😥

**Comment by mpdifran** [comment]
From what I remember we employed some hacks to get that to work. Perhaps newer versions of the iOS SDK broke those hacks.

**Comment by yoichitgy** [comment]
It looks the cause relate to the new NSStoryboard APIs Apple introduced for macOS 10.15+ and Xcode 11.0+

- The old and still available API is `instantiateInitialController()` https://developer.apple.com/documentation/appkit/nsstoryboard/1426545-instantiateinitialcontroller
- The new API to instantiate NSViewController is `instantiateInitialController(creator:)` https://developer.apple.com/documentation/appkit/nsstoryboard/3203779-instantiateinitialcontroller
- The new API to instantiate NSWindowController is `instantiateInitialController(creator:)` https://developer.apple.com/documentation/appkit/nsstoryboard/3203780-instantiateinitialcontroller

I'm still investigating the issue.

**Comment by yoichitgy** [comment]
I made the investigation, but still not sure why `instantiateInitialController` doesn't work for macOS target...

**Comment by mpdifran** [comment]
@yoichitgy release has been completed! Feel free to merge when you're ready!

**Comment by yoichitgy** [comment]
Thanks @mpdifran for the review😃

I made a ticket #166 for the macOS issue found during the work for this PR.

**Pull request merged**